tools/calyx-writer/lib.rs

```diff
@@ -789,14 +800,20 @@ impl Component {
             }
             assert!(!port.has_inferred_width());
             port.attributes.write(f)?;
-            write!(f, "{}: {}", port.name, port.width)?;
+            write!(
+                f,
+                "{}: {}",
+                port.name,
+                port.width
+                    .expect("we just checked that port has concrete width")
```

Comment:
It's fine to have this be an `unwrap` since you've covered it in the assert. Or you can remove the assert and make the error message a bit more clear and keep it as an expect